src/main/java/org/opensearch/timeseries/util/ParseUtils.java

```diff
@@ -688,6 +696,46 @@ public static String checkFilterByBackendRoles(User requestedUser) {
         return null;
     }
 
+    /**
+     * Verifies whether the user has permission to access the resource.
+     */
+    public static void verifyResourceAccessAndProcessRequest(
+        User requestedUser,
+        String detectorId,
+        NodeClient client,
+        Settings settings,
+        Version nodeVersion,
+        ActionListener<? extends ActionResponse> listener,
+        Consumer<Object[]> onSuccess,
+        Object[] successArgs,
+        BiConsumer<Throwable, Object[]> fallbackOn501,
+        Object[] fallbackArgs
+    ) {
+        ResourceSharingClient resourceSharingClient = ResourceSharingClientAccessor.getResourceSharingClient(client, settings, nodeVersion);
```

Comment:
We can have a separate getter method for the client

Reply:
I'm confused. The ResourceSharingClientAccessor class already defines a separate getter method.